### DISCUSSION TITLE
define_asset_job config and accessing these details in an asset

### DISCUSSION THREAD

**Discussion by aa-odim** [dagster-io/dagster#20299] [Q&A]
I'm trying to pass configuration (dates) to assets so that I can override these values in the UI when needed. 

Specifically I have a series of assets that form a dependency stream and would each need access to the configuration details. The assets are collected into a job using define_asset_job and this job is executed on a schedule via a ScheduleDefinition. The configuration I am trying to pass is a subclass of Config (shown below)

The approach I used is to define a `RunConfig` for the `config` parameter of `define_asset_job`

So as an example: `asset_1` -> `asset_2` -> `asset_3`

```python
class MyConfig(Config):
    start_dt: str = (datetime.today() - timedelta(4)).strftime("%Y-%m-%d")
    end_dt: str = datetime.today().strftime("%Y-%m-%d")

asset_job = define_asset_job(
    AssetSelection.keys("asset_1", "asset_2", "asset_3"), 
    config=RunConfig(ops={"asset_1": MyConfig(), "asset_2": MyConfig(), "asset_3": MyConfig()})
)


@asset
def asset_1(config: MyConfig):
    # some logic

@asset(deps=[asset_1])
def asset_2(config: MyConfig):
    # some logic

@asset(deps=[asset_2])
def asset_3(config: MyConfig):
    # some logic
```

The problem is twofold: 
1. I can't alter the configuration in the UI since this is a define_asset_job 
2. I need the altered configuration to apply to all the assets that materialize, not just the asset i alter

Is there a better approach to this?

**Comment by alangenfeld** [comment]
this is a relevant discussion for having config that applies across an entire job 

https://github.com/dagster-io/dagster/discussions/3213
